fbw-a32nx/src/systems/shared/src/NavigationDisplay.ts

```diff
@@ -2,11 +2,20 @@
 //  SPDX-License-Identifier: GPL-3.0
 
 import { Coordinates } from '@fmgc/flightplanning/data/geo';
+import { AircraftType } from '@shared/aircraft';
 
 export type EfisSide = 'L' | 'R'
 
-export type EfisNdRangeValue = 10 | 20 | 40 | 80 | 160 | 320;
-export const rangeSettings: EfisNdRangeValue[] = [10, 20, 40, 80, 160, 320];
+export type EfisNdRangeValue = 0.5 | 10 | 20 | 40 | 80 | 160 | 320 | 640;
+export const rangeSettingValues: EfisNdRangeValue[] = [0.5, 10, 20, 40, 80, 160, 320, 640];
+
+export const getRangeSettingsValue = (aircraft: AircraftType, index: number): EfisNdRangeValue => {
```

Comment:
I think I would prefer that we just change the A380 modelbehaviors to use the same indices for the range settings that do overlap, and negative values for ZOOM ranges. Passing around an aircraft type isn't an idea I like a lot

Reply:
Besides we're doing OANS on A320 anyway, and headwind are too, so we need support on both.

Reply:
I don't like to map the A380 modelbehaviors of the A380 and have special values for zoom, etc.
This is not really clean for the underlying systems (EGPWC).

I like tracernz's solution much more to use aircraft specific files during compile time.
Would require to move the ND to fbw-common and special implementations for fbw-a320 and fbw-a380.
Is the cleaner solution in the end.